cmd/kube-apiserver/app/options/options.go

```diff
@@ -72,23 +72,57 @@ type ServerRunOptions struct {
 	EndpointReconcilerType string
 
 	ServiceAccountSigningKeyFile string
+	Storage                      *StorageFactoryOptions
+}
+
+// NewStorageFactoryOptions creates a default option for storage factory
+func NewStorageFactoryOptions() *StorageFactoryOptions {
+	return &StorageFactoryOptions{
+		Etcd:                 genericoptions.NewEtcdOptions(storagebackend.NewDefaultConfig(kubeoptions.DefaultEtcdPathPrefix, nil)),
+		StorageSerialization: kubeoptions.NewStorageSerializationOptions(),
+	}
+}
+
+// StorageFactoryOptions defines behavior of storage factory
+type StorageFactoryOptions struct {
```

Comment:
Rather than combine the two like this, is it possible to instead have the `StorageSerializationOptions` have a method to create the `storageFactory` which is then passed to the `etcdOptions.Apply` method?  That would force us to consider how to cleanly decouple the sequence of creating runnable objects instead of encouraging us to keep these units together

Reply:
thanks @deads2k ! WDYT now?